tests/Doctrine/Tests/DBAL/Functional/NamedParametersTest.php

```diff
@@ -212,6 +214,63 @@ public function testTicket($query,$params,$types,$expected)
             $result[$k] = array_change_key_case($v, CASE_LOWER);
         }
 
-        self::assertEquals($result, $expected);
+        self::assertEquals($expected, $result);
+    }
+
+    public function namedParametersAfterEscapeProvider()
+    {
+        return [
+            ["SELECT 1 FROM ddc1372_foobar WHERE (:param_0 LIKE :find ESCAPE '{{escape}}') OR (:param_1 LIKE :find ESCAPE '{{escape}}') LIMIT 1", false],
+            ['SELECT 1 FROM ddc1372_foobar WHERE (:param_0 LIKE :find ESCAPE \'{{escape}}\') OR (:param_1 LIKE :find ESCAPE "{{escape}}") LIMIT 1', true],
+            ['SELECT 1 FROM ddc1372_foobar WHERE (:param_0 LIKE :find ESCAPE "{{escape}}") OR (:param_1 LIKE :find ESCAPE \'{{escape}}\') LIMIT 1', true],
+            ['SELECT 1 FROM ddc1372_foobar WHERE (:param_0 LIKE :find ESCAPE "{{escape}}") OR (:param_1 LIKE :find ESCAPE "{{escape}}") LIMIT 1', true],
+        ];
+    }
+
+    /**
+     * @dataProvider namedParametersAfterEscapeProvider
+     * @param string $query
+     * @param bool $ignorePostgres
+     */
+    public function testNamedParametersAfterEscapeFirstMatches($query, $ignorePostgres)
```

Comment:
The functional test should only only cover the literals quoted according to the platform rules: `AbstractPlatform::quoteStringLiteral()`. The fact that the parser supports other quote characters than the current platform supports will be covered in the unit test for the parser.